File: src/main/java/SimpleFlowable.java
import io.reactivex.BackpressureStrategy;
import io.reactivex.Flowable;
import io.reactivex.schedulers.Schedulers;
import org.reactivestreams.Subscriber;
import org.reactivestreams.Subscription;

public class SimpleFlowable {
    public static void main(String[] args) throws Exception {
        Flowable<String> flow = Flowable.create(emitter -> {
            System.out.println(
                Thread.currentThread().getName() + ":start sending"
            );

            String[] strings = {"f001", "f002", "f003", "f004", "f005"};

            for (String string : strings) {
                if (emitter.isCancelled()) {
                    return;
                }
                emitter.onNext(string);
            }

            emitter.onComplete();
        }, BackpressureStrategy.BUFFER);

        flow.observeOn(Schedulers.computation()).subscribe(new Subscriber<String>() {
            private Subscription subscription;

            @Override
            public void onSubscribe(Subscription s) {
                System.out.println(
                    Thread.currentThread().getName() + ": Flowable subscribe"
                );
                subscription = s;
                subscription.request(1L);
            }

            @Override
            public void onNext(String s) {
                System.out.println(
                    Thread.currentThread().getName() + ": " + s
                );
                subscription.request(1L);
            }

            @Override
            public void onError(Throwable t) {
                t.printStackTrace();
            }

            @Override
            public void onComplete() {
                System.out.println(
                    Thread.currentThread().getName() + ": onComplete"
                );
            }
        });

        Thread.sleep(500L);
    }
}

File: src/main/kotlin/SimpleFlowable.kt
import io.reactivex.*
import io.reactivex.schedulers.Schedulers
import org.reactivestreams.Subscriber
import org.reactivestreams.Subscription
import kotlin.system.exitProcess

fun main() {
    val flow = Flowable.create<String>({ emitter ->
        println(
            Thread.currentThread().name + ":start sending"
        )

        val strings = arrayOf("f001", "f002", "f003", "f004", "f005")

        for (string in strings) {
            if (emitter.isCancelled) {
                exitProcess(0)
            }
            emitter.onNext(string)
        }

        emitter.onComplete()
    }, BackpressureStrategy.BUFFER)

    flow.observeOn(Schedulers.computation()).subscribe(object : Subscriber<String> {
        private var subscription: Subscription? = null

        override fun onSubscribe(s: Subscription) {
            println(
                Thread.currentThread().name + ": Flowable subscribe"
            )
            subscription = s
            subscription!!.request(1L)
        }

        override fun onNext(s: String) {
            println(
                Thread.currentThread().name + ": " + s
            )
            subscription!!.request(1L)
        }

        override fun onError(t: Throwable) {
            t.printStackTrace()
        }

        override fun onComplete() {
            println(
                Thread.currentThread().name + ": onComplete"
            )
        }
    })

    Thread.sleep(500L)
}
